lib/util/conflicter.js

```diff
@@ -165,7 +171,31 @@ Conflicter.prototype._ask = function (file, cb) {
     }
 
     if (result.action === 'diff') {
-      this.adapter.diff(fs.readFileSync(file.path, 'utf8'), file.contents.toString());
+      var existingHeader = readChunk.sync(file.path, 0, 512);
+
+      if (istextorbinary.isBinarySync(undefined, existingHeader) || istextorbinary.isBinarySync(undefined, file.contents)) {
```

Comment:
Move the content of the if statement to a helper function (e.g. `diffBinary`). It'll be easier to unit test, and easier to read.